edgelet/hsm-sys/azure-iot-hsm-c/src/edge_hsm_client_store.c

```diff
@@ -2676,8 +2676,7 @@ static int edge_hsm_client_store_insert_encryption_key
         LOG_DEBUG("HSM store already has encryption key set %s", key_name);
         result = 0;
     }
-    else if ((load_encryption_key_from_file(g_crypto_store, key_name) == 0) &&
-                key_exists((CRYPTO_STORE*)handle, HSM_KEY_ENCRYPTION, key_name))
+    else if (load_encryption_key_from_file(g_crypto_store, key_name) == 0)
```

Comment:
Isn't it possible that the key with the given name does not exist even after the `load_encryption_key_from_file` call?

Reply:
Never mind. That's not possible.